lib/puppetserver/utils/http_client.rb

```diff
@@ -45,6 +45,13 @@ def initialize(net_http_connection, url_struct)
           @url = url_struct
         end
 
+        def get(url_overide = nil)
+          url = url_overide || @url
+
+          request = Net::HTTP::Get.new(url.to_uri, HEADERS)
```

Comment:
@Magisus could our auth problem have been that this method lacked "HEADERS" ?

Reply:
I didn't even try this one. I did sign (which is a put) and clean (which is a delete).

Reply:
I don't see how
```
HEADERS = {
        'User-Agent'   => 'PuppetserverCaCli',
        'Content-Type' => 'application/json',
        'Accept'       => 'application/json'
      }
```
would matter but ?

Reply:
Yeah, the get only gets called with the --all flag i guess